xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultRefactoringConfiguration.java

```diff
@@ -45,9 +43,10 @@
     private ConfigurationSource configurationSource;
 
     @Override
-    public boolean canSkipRecycleBin()
+    public boolean isRecycleBinSkippingActivated()
     {
-        String canSkipRecycleBin = this.configurationSource.getProperty("canSkipRecycleBin", "0");
-        return Objects.equals("1", canSkipRecycleBin);
+        Boolean isRecycleBinSkippingActivated =
+            this.configurationSource.getProperty("isRecycleBinSkippingActivated", false);
+        return isRecycleBinSkippingActivated != null && isRecycleBinSkippingActivated;
```

Comment:
why check for null since you use a default?

Reply:
you should be able to use a `boolean`.

Reply:
Now using `boolean` https://github.com/xwiki/xwiki-platform/pull/1401/files#diff-3c76751dca65c1a50961704e4a11dd04R48

I had an NPE previously, because of a invalid use of Mockito, that was not reflecting the way `getProperty` is actually working.